caddyhttp/browse/setup.go

```diff
@@ -477,6 +477,16 @@ footer {
 			var filterEl = document.getElementById('filter');
 			filterEl.focus();
 
+			function initFilter() {
+				if (!filterEl.value) {
+					var filterParam = new URL(window.location.href).searchParams.get('filter');
+					if (filterParam) {
+						filterEl.value = filterParam;
+					}
+				}
+				filter();
```

Comment:
`filter()` needs to be executed unconditionally on load to preserve the behavior added in #2368.